versioning/BRM/extentmap.cpp

```diff
@@ -6164,6 +5894,43 @@ void ExtentMap::finishChanges()
         releaseEMEntryTable(WRITE);
 }
 
+void ExtentMap::makeUndoRecordRBTree(UndoRecordType type, const EMEntry& emEntry)
+{
+    undoRecordsRBTree.push_back(make_pair(type, emEntry));
+}
+
+void ExtentMap::undoChangesRBTree()
```

Comment:
I am curios if undo code takes emindex into account? What happens if one removes/adds EMEntry so that it is removed/added from/into the index and rollback then?

Reply:
Does it have support in original patch?

Reply:
No, it doesn't.